.github/workflows/ci.yml

```
@@ -0,0 +1,62 @@
+# This pipe line will be triggered when a pull request is created and will validiate that the CUDA and OpenCL versions of Autoscoper can configure and build successfully.
+
+name: CI - Ubuntu
+
+on:
+  pull_request:
+    branches: [ main ]
+
+
+env: 
+  BUILD_TYPE: Release
+  QT_VERSION:   5.15.2
+
+jobs:
+  
+  build-cuda:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Install Qt
+      uses: jurplel/install-qt-action@v3
+      with:
+        version:      ${{ env.QT_VERSION }}
+        host:         linux
+        target:       desktop
+        arch:         gcc_64
+        dir:          /opt/qt
+        modules:      webassembly qtwebengine sources qtwebglplugin qtscript debug_info
+        setup-python: false
+    - name: Install CUDA
```

```suggestion
    - name: Install CUDA
      if: ${{ matrix.rendering_backend == 'CUDA' }}
      uses: Jimver/cuda-toolkit@v0.2.8
```

This assumes the matrix will be defined using something like this


```yaml
jobs:
  autoscoper:
    strategy:
      matrix:
        rendering_backend: ['CUDA', 'OpenCL']
    steps:
      ....
```